packages/dev/core/src/Gizmos/planeRotationGizmo.ts

```diff
@@ -181,24 +202,32 @@ export class PlaneRotationGizmo extends Gizmo implements IPlaneRotationGizmo {
         this._coloredMaterial.specularColor = color.subtract(new Color3(0.1, 0.1, 0.1));
 
         this._hoverMaterial = new StandardMaterial("", gizmoLayer.utilityLayerScene);
-        this._hoverMaterial.diffuseColor = Color3.Yellow();
+        this._hoverMaterial.diffuseColor = hoverColor;
+        this._hoverMaterial.specularColor = hoverColor;
 
         this._disableMaterial = new StandardMaterial("", gizmoLayer.utilityLayerScene);
-        this._disableMaterial.diffuseColor = Color3.Gray();
+        this._disableMaterial.diffuseColor = disableColor;
         this._disableMaterial.alpha = 0.4;
 
         // Build mesh on root node
         this._gizmoMesh = new Mesh("", gizmoLayer.utilityLayerScene);
         const { rotationMesh, collider } = this._createGizmoMesh(this._gizmoMesh, thickness, tessellation);
 
         // Setup Rotation Circle
-        this._rotationDisplayPlane = CreatePlane("rotationDisplay", { size: 0.6, updatable: false }, this.gizmoLayer.utilityLayerScene);
+        this._rotationDisplayPlane = CreatePlane(
+            "rotationDisplay",
+            {
+                size: 0.6,
+                updatable: false,
+            },
+            this.gizmoLayer.utilityLayerScene
+        );
         this._rotationDisplayPlane.rotation.z = Math.PI * 0.5;
         this._rotationDisplayPlane.parent = this._gizmoMesh;
         this._rotationDisplayPlane.setEnabled(false);
 
         Effect.ShadersStore["rotationGizmoVertexShader"] = PlaneRotationGizmo._RotationGizmoVertexShader;
-        Effect.ShadersStore["rotationGizmoFragmentShader"] = PlaneRotationGizmo._RotationGizmoFragmentShader;
+        Effect.ShadersStore["rotationGizmoFragmentShader"] = PlaneRotationGizmo._createRotationGizmoFragmentShader(hoverColor);

```

Comment:
I was thinking about this a bit. Setting the shader as a fixed string will pevent it from being reused with different colors.
setting the color as a uniform to the shader material will allow multiple instances of this to use different colors

Reply:
I second Raanan. Standard way is to use uniform, not to change the shader code.

Reply:
the solution with adding a floating point will also be solved if you do that.

Reply:
Sure! Check this line in the shader: `uniform vec3 angles;`  It's a uniform constant that declared in the shader and set on the CPU side in 2 steps:
- declaring it here : https://github.com/BabylonJS/Babylon.js/blob/f7793c25a456b56801e6b352287341b0a1d4525c/packages/dev/core/src/Gizmos/planeRotationGizmo.ts#L240
- and set here : https://github.com/BabylonJS/Babylon.js/blob/f7793c25a456b56801e6b352287341b0a1d4525c/packages/dev/core/src/Gizmos/planeRotationGizmo.ts#L278

Basically, you have to do the same thing with diffuse colors instead of angles.

Reply:
Thanks for the hint!

I think/hope I got it :tada:.